lint/highlight.py

```diff
@@ -387,47 +430,82 @@ def draw(self, view):
+                scope = self.get_style("scope", style, error_type)
+                mark_style = self.get_style("mark_style", style, error_type)
+
+                flags = MARK_STYLES[mark_style]
+                view.add_regions(style, regions, scope=scope, flags=flags)
+                drawn_regions.append(style)
+
+            # gutter marks
+            if not persist.has_gutter_theme:
+                continue
+
+            gutter_regions = {}
+            # collect regions of error type
+            for line, style in self.lines[error_type].items():
+                pos = self.newlines[line]
+                region = sublime.Region(pos, pos)
+                gutter_regions.setdefault(style, []).append(region)
+
+            # draw gutter marks for
+            for style, regions in gutter_regions.items():
+                icon = self.get_style("icon", style, error_type)
+                if icon == "none":  # do not draw icon
+                    continue
+
+                # colorize icon
+                if (persist.gutter_marks['colorize'] or icon in ST_ICONS):
+                    scope = self.get_style("scope", style, error_type)
                 else:
-                    scope = 'sublimelinter.gutter-mark'
+                    scope = " "  # set scope to non-existent one
+
+                k = style.rfind(".")
+                gutter_key = style[:k] + ".gutter." + style[k+1:]
 
+                # GUTTER_MARK_KEY_FORMAT.format(error_type)
                 view.add_regions(
-                    GUTTER_MARK_KEY_FORMAT.format(error_type),
-                    gutter_regions[error_type],
-                    scope,
-                    icon=persist.gutter_marks[error_type]
+                    gutter_key,
+                    regions,
+                    scope=scope,
+                    icon=icon
                 )
+                drawn_regions.append(gutter_key)
+                protected_regions.extend(regions)
+
+            # overlaying all gutter regions with common invisible one,
+            # to create unified handle for GitGutter and other plugins
+            # flag might not be neccessary
+            view.add_regions(PROTECTED_REGIONS_KEY, protected_regions, flags=sublime.HIDDEN)
+            drawn_regions.append(PROTECTED_REGIONS_KEY)
+
+            # persisting region keys for later clearance
+            persist.region_store.add_region_keys(view, drawn_regions)
 
     @staticmethod
     def clear(view):
         """Clear all marks in the given view."""
-        for error_type in (WARNING, ERROR):
-            view.erase_regions(MARK_KEY_FORMAT.format(error_type))
-            view.erase_regions(GUTTER_MARK_KEY_FORMAT.format(error_type))
+        # # TODO: clear all regions
+        # for error_type in (WARNING, ERROR):
+        #     view.erase_regions(MARK_KEY_FORMAT.format(error_type))
+        #     view.erase_regions(GUTTER_MARK_KEY_FORMAT.format(error_type))
+        persist.region_store.del_regions(view)
```

Comment:
If `clear` only calls another function, just replace it everywhere.

Reply:
We can refactor this another time. For now, I have just removed the comments.